src/timepicker/TimePickerState.ts

```diff
@@ -50,19 +50,19 @@ export const useTimePickerState = (props: TimePickerStateProps = {}) => {
     time?.getHours() >= 12 ? "PM" : "AM",
   );
 
-  const setHour = (hour: number) => {
-    setHourProp(hour);
-    setTime(hour, minute, meridian);
+  const setHour = (hour: number | string) => {
+    setHourProp(hour as number);
+    setTime(hour as number, minute, meridian);
   };
 
-  const setMinute = (minute: number) => {
-    setMinuteProp(minute);
-    setTime(hour, minute, meridian);
+  const setMinute = (minute: number | string) => {
+    setMinuteProp(minute as number);
+    setTime(hour, minute as number, meridian);
   };
 
-  const setMeridian = (meridianValue: string) => {
-    setMeridianProp(meridianValue);
-    setTime(hour, minute, meridianValue);
+  const setMeridian = (meridianValue: number | string) => {
+    setMeridianProp(meridianValue as string);
+    setTime(hour, minute, meridianValue as string);
```

Comment:
Hmmmm 🤔 this works but i think we can utilize discriminated union functions on the onChange type

```ts
  onChange?: ((v: number) => void) | ((v: string) => void);
```

Reply:
I just fixed it temporarily. We need to fix it by providing better onChange types.

Need you help, I am not as good as you in TS.

Reply:
Yup thats what we want, better onChange, this should work.

onChange?: ((v: number) => void) | ((v: string) => void);